lib/model/autocomplete.dart

```diff
@@ -483,6 +468,42 @@ class MentionAutocompleteView extends AutocompleteView<MentionAutocompleteQuery,
     return userAName.compareTo(userBName); // TODO(i18n): add locale-aware sorting
   }
 
+  bool _isChannelWildcardIncluded = false;
+
+  @override
+  Future<List<MentionAutocompleteResult>?> computeResults() async {
+    _isChannelWildcardIncluded = false;
+    final results = <MentionAutocompleteResult>[];
+    // give priority to wildcard mentions
+    if (await filterCandidates(filter: _testWildcard,
```

Comment:
The fancy `filterCandidates` machinery isn't needed for the wildcards, because there's only a handful of them. (It's useful when there are potentially thousands of candidates.) So this can be simplified to a direct loop through the wildcard candidates.

That also lets `_inChannelWildcardIncluded` be a local variable, which simplifies reasoning about it.

… Then I started writing that the logic might be further clarified by unrolling the loop, akin to the logic in zulip-mobile:
https://github.com/zulip/zulip-mobile/blob/715d60a5e87fe37032bce58bd72edb99208e15be/src/autocomplete/WildcardMentionItem.js#L113-L138

But I see we discussed this previously above:
https://github.com/zulip/zulip-flutter/pull/889#discussion_r1727788259
and that you have the corresponding logic living instead in the widgets code, in `ComposeAutocomplete._wildcards`. That structure may be fine too. Should have a link to the zulip-mobile version, though, as discussed there.